### PR TITLE
Allow filtering of log messages based on severity level

### PR DESCRIPTION
**Pull request by dalebremner**
@benbp @jlongever @brianparry

### CODE REVIEW

File: spec/lib/common/logger-spec.js [hunk 1]
```diff
@@ -6,13 +6,14 @@
...
 describe('Logger', function () {
-    var Logger, events;
+    var Logger, events, configuration;
 
     helper.before();
 
     before(function () {
         Logger = helper.injector.get('Logger');
         events = helper.injector.get('Events');
+        configuration = helper.injector.get('Services.Configuration');
```
Comment (benbp) [inline spec/lib/common/logger-spec.js]: Ignoreable.

File: lib/common/logger.js [hunk 2]
```diff
@@ -57,6 +59,13 @@ function loggerFactory(
         assert.isIn(level, levels);
         assert.string(message, 'message');
 
+        // Exit if the log level of this message is less than the minimum log level.
+        var minLogLevel = nconf.get('minLogLevel');
+        if ((minLogLevel === undefined) || (typeof minLogLevel !== 'number')) {
+            minLogLevel = 0;
+        }
+        if (Constants.Logging.Levels[level] < minLogLevel) return;
```
Comment (benbp) [inline lib/common/logger.js]: We try to avoid braceless if statements as a style rule, if you wouldn't mind changing this.

**Comment by benbp** [comment]
:+1: thanks!

**Comment by iceiilin** [comment]
May I ask is there any related PRs in other repos? Or how will this filter be used?

**Comment by yyscamper** [comment]
:+1:

**Comment by dalebremner** [comment]
Hi Lyne. You will be able to limit the verbosity of log messages emitted by monorail by specifying the minimum level in monorail.json

**Comment by heckj** [comment]
@dalebremner quick question - if I wanted to add a stanza to filter on warning of worse, then I'd need to add 

    minLogLevel: 4

to the monorail.json configuration file? Adding quick docs in https://github.com/RackHD/docs/pull/212 and wanted to verify I had it correct

**Comment by benbp** [comment]
Ah I hadn't thought about that. Would be good if we could do conversion so once could specify by log level name rather than just number.

**Comment by dalebremner** [comment]
@heckj correct, 4 would be "warning" or worse...

**Pull request merged**